File: backend/insurance_engine/src/InsuranceEngine.Policy/Domain/Entities/Nominee.cs
using InsuranceEngine.SharedKernel.Domain;
using InsuranceEngine.SharedKernel.Domain.ValueObjects;

namespace InsuranceEngine.Policy.Domain.Entities;


public class Nominee : Entity<Guid>
{
    public Nominee(Guid id) : base(id) { }
    public Nominee() { }
    public Guid PolicyId { get; set; }
    public Guid? BeneficiaryId { get; set; }

    // --- Proto-aligned inline fields ---

    public string FullName { get; set; } = string.Empty;
    public string Relationship { get; set; } = string.Empty;
    public double SharePercentage { get; set; }
    public DateTime? DateOfBirth { get; set; }
    public string? NomineeDobText { get; set; }
    public string? NidNumber { get; set; }
    public string? PhoneNumber { get; set; }

    // Audit
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? DeletedAt { get; set; }
}

File: sdks/insuretech-go-sdk/pkg/models/nominee.go
package models

import (
	"time"
)

// Nominee represents a nominee
type Nominee struct {
	CreatedAt time.Time `json:"created_at,omitempty"`
	DateOfBirth time.Time `json:"date_of_birth,omitempty"`
	FullName string `json:"full_name,omitempty"`
	NidNumber string `json:"nid_number,omitempty"`
	NomineeDobText string `json:"nominee_dob_text,omitempty"`
	NomineeId string `json:"nominee_id,omitempty"`
	NomineeSharePercent float64 `json:"nominee_share_percent,omitempty"`
	PhoneNumber string `json:"phone_number,omitempty"`
	PolicyId string `json:"policy_id,omitempty"`
	Relationship string `json:"relationship,omitempty"`
	SharePercentage float64 `json:"share_percentage,omitempty"`
	UpdatedAt time.Time `json:"updated_at,omitempty"`
}
